File: code_samples/csharp/functional_monads.cs
using System;

/// <summary>
/// Functional: Monads
/// AI/ML Training Sample
/// </summary>
public class Monads
{
    private string data;
    
    public Monads()
    {
        this.data = string.Empty;
    }
    
    public void Process(string input)
    {
        this.data = input;
    }
    
    public string GetData()
    {
        return this.data;
    }
    
    public bool Validate()
    {
        return !string.IsNullOrEmpty(this.data);
    }
    
    public static void Main(string[] args)
    {
        var instance = new Monads();
        instance.Process("example");
        Console.WriteLine($"Data: {instance.GetData()}");
        Console.WriteLine($"Valid: {instance.Validate()}");
    }
}

File: code_samples/cpp/functional_monads.cpp
/**
 * Functional: Monads
 * AI/ML Training Sample
 */

#include <iostream>
#include <string>

class Monads {
private:
    std::string data;
    
public:
    Monads() : data("") {}
    
    void process(const std::string& input) {
        data = input;
    }
    
    std::string getData() const {
        return data;
    }
    
    bool validate() const {
        return !data.empty();
    }
};

int main() {
    Monads instance;
    instance.process("example");
    std::cout << "Data: " << instance.getData() << std::endl;
    std::cout << "Valid: " << instance.validate() << std::endl;
    return 0;
}
